examples/misc/keyword_plotting.py

```diff
@@ -0,0 +1,24 @@
+"""
+Plotting with keywords
+======================
+
+There are some instances where you have data in a format that lets you
+access particular variables with strings. For example, with
+:class:`numpy.recarray` or :class:`pandas.DataFrame`.
+
+Matplotlib allows you provide such an object with
+the ``data`` keyword argument. If provided, then you may generate plots with
+the strings corresponding to these variables.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+np.random.seed(19680801)
+
+data = dict([(label, np.random.randn(50)) for label in ['a', 'b', 'c', 'd']])
+data['d'] = np.abs(data['d']) * 100
```

Comment:
Hmm, as much as it's more work in some ways, I feel like an example with a clear structure, so something like:
```python
data = { 'a' : [1,2,3,4,5,5,6,7],
              'b' :[4,5, 6,2,3, 4, 5],
              'c': [3,1,2,3,4,5,6,3],
              'd':[1,2,3,4,5,6,7,8]}
```
might work better since the point of this is to really clearly see what the kwarg does.